rllib/models/modelv2.py

```diff
@@ -439,7 +439,10 @@ def _unpack_obs(obs: TensorType, space: Space, tensorlib: Any = tf) -> TensorStr
             )
         offset = 0
         if tensorlib == tf:
-            batch_dims = [v if isinstance(v, int) else v.value for v in obs.shape[:-1]]
+            batch_dims = [
+                v if isinstance(v, (int, type(None))) else v.value
```

Comment:
thanks for the fix.
personally isinstance(v, type(None)) feels a bit weird, maybe you can simply:
```
[v if v is None or isinstace(v, int) else v.value for v in obs.shape[:-1]]
```
now considering the very next statement, feel like we can do both with a simple inline fn:
```
def get_v(v):
    if v is None: return -1
    elif isinstance(v, int): return v
    else return v.value
batch_dims = [get_v(v) for v in obs.shape[:-1]]
```

Reply:
Changed the on liner to the second method in cbb234567fa42d4aa412b43c13af3b8e33e175b1